src/satosa/backends/saml2.py

```diff
@@ -248,8 +320,11 @@ def authn_response(self, context, binding):
             raise SATOSAAuthenticationError(context.state, "State did not match relay state")
 
         context.decorate(Context.KEY_BACKEND_METADATA_STORE, self.sp.metadata)
-
-        del context.state[self.name]
+        if self.config.get(SAMLBackend.KEY_MEMORIZE_IDP):
```

Comment:
No need to actually check for this - it does not matter how we store, but how/when we use it.